src/Mvc/Mvc.Analyzers/src/CodeAnalysisExtensions.cs

```diff
@@ -141,11 +141,12 @@ public static bool HasAttribute(this ISymbol symbol, ITypeSymbol attribute)
 
         private static IEnumerable<ITypeSymbol> GetTypeHierarchy(this ITypeSymbol typeSymbol)
         {
-            while (typeSymbol != null)
+            var typeSymbolNullable = typeSymbol;
```

Comment:
These changes are to fix some nullability errors during the build. Could use some eyes on these.

Reply:
Who on our team has done nullability work? @safern do you know?

Reply:
I and @buyaa-n mostly. Looking at the change now. @JunTaoLuo what was the error you were getting, could you share?

Reply:
If the method parameter wouldn't be nullalbe we normally suppress warning for `typeSymbol = typeSymbol.BaseType!;` as loop would stop on null `while (typeSymbol != null)` and there is no risk of NRE

Reply:
Right, also the most natural thing is to mark typeSymbol as nullable:

```cs
private static IEnumerable<ITypeSymbol> GetTypeHierarchy(this ITypeSymbol? typeSymbol)
{
    ...
}
```

Since you're checking for null you'll be safe, never returning null and never null ref and you'll get no warnings:

https://sharplab.io/#v2:CYLg1APgxAdgrgGwQQwEYIKYAIMzZgWACgABAJgEZiSAGLEigFgG5jqBmeigNnrKwDCWAN7Es4+pwa8G7ADwBJAGIB7FQD4sAWQAUAFwAWASwDOWZWoD8WAG7IEcDAEoxE0UQmesAd2OYsOnYO2ACEALxY8EguHl7i7nFxDBT0AOy29o6ssYkSrrniQY5YEUUYAHQAQsgmGNm5AL75WE1ErRxYRjB6GABOAGbIAMbYFirECRIknGPW1bUiWADmGHrMWLVrLcQNQA

Reply:
I'll give this a shot

Reply:
@dougbu Yes these came up after the new roslyn versions were ingested. Since there weren't that many of them, I figured I'd just resolve them.

Reply:
Fyi, there are about 5 other files affected by nullability errors that I made changes to.

Reply:
It might be good if you could put those in a single commit to make review easier... I'll be happy to review the nullability changes

Reply:
I'll just list the files here:
https://github.com/dotnet/aspnetcore/pull/21581/files#diff-ba3993cca35ae5a2f85d960dcfd8145b
https://github.com/dotnet/aspnetcore/pull/21581/files#diff-c3196cc6809256e56566396bcad09236
https://github.com/dotnet/aspnetcore/pull/21581/files#diff-d84be6008957ad062279c6fa1eba3fcb
https://github.com/dotnet/aspnetcore/pull/21581/files#diff-2dd827d6f225bc54fc487852fd6fd7bc
https://github.com/dotnet/aspnetcore/pull/21581/files#diff-affae284d09d0a117f05ed0310f4ccd3
https://github.com/dotnet/aspnetcore/pull/21581/files#diff-c35f544d1d37d5777f4e1ee18fae4dc5